lib/hammer_cli_foreman/api/connection.rb

```diff
@@ -107,6 +112,14 @@ def self.foreman_api_reconnect(auth_type)
     end
   end
 
+  def self.previous_session_auth(settings)
+    return nil unless settings.get(:foreman, :use_sessions)
+
+    url = settings.get(:_params, :host) || settings.get(:foreman, :host)
+    session = HammerCLIForeman::Sessions.get(url)
+    session.auth_type if session.valid? && session.expired?
+  end
```

Comment:
This method is used only here, so we can change it to
```ruby
def self.previous_session_auth
  settings = HammerCLI::Settings
  #... same code
end
```

If you agree, don't forget to adjust line 94.